designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/CustomActionValidator.scala

```diff
@@ -0,0 +1,160 @@
+package pl.touk.nussknacker.ui.validation
+
+import cats.data.Validated.Invalid
+import io.circe.{Decoder, Encoder, HCursor, Json}
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.context.PartSubGraphCompilationError
+import pl.touk.nussknacker.engine.api.deployment.CustomActionCommand
+import pl.touk.nussknacker.engine.api.parameter.ParameterName
+import pl.touk.nussknacker.engine.deployment.{
+  CustomActionDefinition,
+  CustomActionParameter,
+  CustomActionValidationResult
+}
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.restmodel.CustomActionRequest
+import pl.touk.nussknacker.ui.BadRequestError
+
+class CustomActionValidator(val allowedActions: List[CustomActionDefinition]) {
+
+  def validateCustomActionParams(
+      request: CustomActionRequest
+  ): Either[CustomActionValidationError, CustomActionValidationResult] = {
+
+    val checkedCustomAction =
+      allowedActions
+        .find(_.name == request.actionName)
+        .toRight(CustomActionNonExistingError("Couldn't find this action: " + request.actionName.value))
+
+    checkedCustomAction match {
+      case Left(notFoundAction) => Left(notFoundAction)
+
+      case Right(foundAction) => {
+        implicit val nodeId: NodeId = NodeId(foundAction.name.value)
+        val customActionParams      = foundAction.parameters
+        val requestParamsMap        = getRequestParamsMap(request, customActionParams)
+
+        val validated = validateParams(requestParamsMap, customActionParams)
+        getValidationResult(validated)
+      }
+    }
+  }
+
+  private def getRequestParamsMap(request: CustomActionRequest, customActionParams: List[CustomActionParameter]) = {
```

Comment:
See similar validation in `ScenarioPropertiesValidator`, it seems that here you want do `getMissingRequiredPropertyValidationResults` and `getUnknownPropertyValidationResults`.

Reply:
added a checkForMissingParameters